ui/components/app/snaps/tx-insight-warnings/tx-insight-warnings.js

```diff
@@ -0,0 +1,199 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Modal,
+  ModalContent,
+  ModalOverlay,
+  ModalHeader,
+  IconName,
+  AvatarIcon,
+  AvatarIconSize,
+  Text,
+  Checkbox,
+  Box,
+  Button,
+  ButtonSecondarySize,
+  ButtonPrimarySize,
+  ButtonVariant,
+} from '../../../component-library';
+import {
+  AlignItems,
+  BackgroundColor,
+  BlockSize,
+  Display,
+  FlexDirection,
+  IconColor,
+  JustifyContent,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { SnapUIRenderer } from '../snap-ui-renderer';
+import { DelineatorType } from '../../../../helpers/constants/snaps';
+import { stripHttpSchemes } from '../../../../helpers/utils/util';
+
+export default function TxInsightWarnings({
+  warnings,
+  type = 'confirming',
+  origin,
+  onCancel,
+  onSubmit,
+}) {
+  const t = useI18nContext();
+  // Values represent if a warning is collapsed
+  const defaultWarningState = warnings.reduce((warningState, warning, idx) => {
+    warningState[warning.snapId] = idx !== 0;
+    return warningState;
+  }, {});
+
+  const [warningState, setWarningState] = useState(defaultWarningState);
+  const [isChecked, setIsChecked] = useState(false);
+
+  const handleOnChange = () => setIsChecked((state) => !state);
+
+  const handleWarningClick = (snapId) => {
+    let newState = { ...warningState };
+    const nextSnapState = !warningState[snapId];
+    const willClose = nextSnapState;
+    newState[snapId] = nextSnapState;
+    if (!willClose) {
+      newState = { ...defaultWarningState };
+      newState[snapId] = false;
+      newState[warnings[0].snapId] = snapId !== warnings[0].snapId;
+    }
+    setWarningState(newState);
+  };
+
+  const Warnings = () => {
+    const lastWarningIdx = warnings.length - 1;
+    return (
+      <Box className="tx-insights-warnings-modal__content">
+        {warnings.map((warning, idx) => {
+          const { snapId, content } = warning;
+          return (
+            <SnapUIRenderer
+              key={`${snapId}-${idx}`}
+              snapId={snapId}
+              data={content}
+              delineatorType={DelineatorType.Warning}
+              onClick={() => handleWarningClick(snapId)}
+              isCollapsable
+              isCollapsed={warningState[snapId]}
+              boxProps={{ marginBottom: idx === lastWarningIdx ? 0 : 4 }}
+            />
+          );
+        })}
+      </Box>
+    );
+  };
+
+  // move this to an enum that defines the language to be used in this modal on the various
+  // screens that will offer transaction insights, should be indexed on "type".
+  const results = {
+    confirming: { noun: 'confirmation', imperative: 'confirm' },
+    signing: { noun: 'signature', imperative: 'sign' },
+  };
+
+  return (
+    <Modal
```

Comment:
Looks like this has an alignment issue
<img width="345" alt="Screenshot 2023-10-27 at 12 11 45" src="https://github.com/MetaMask/metamask-extension/assets/1561200/4f90f01d-618f-4b0e-8843-58430f40c03c">

Reply:
[5a3398c](https://github.com/MetaMask/metamask-extension/pull/20554/commits/5a3398c82e2e7d68d4b11c5585f7d2ac1bb793ff)